### PR TITLE
Refactor find forks and fix possible bugs that weak permissions check

### PR DESCRIPTION
**Pull request by lunny**
- Move models/GetForks to services/FindForks
- Add doer as a parameter of FindForks to check permissions
- Slight performance optimization for get forks API with batch loading of repository units
- Add tests for forking repository to organizations

### CODE REVIEW

File: tests/integration/repo_fork_test.go [hunk 1]
```diff
@@ -74,3 +79,64 @@ func TestRepoForkToOrg(t *testing.T) {
 	_, exists := htmlDoc.doc.Find(`a.ui.button[href*="/fork"]`).Attr("href")
 	assert.False(t, exists, "Forking should not be allowed anymore")
 }
+
+func TestForkListLimitedAndPrivateRepos(t *testing.T) {
+	forkItemSelector := ".tw-flex.tw-items-center.tw-py-2"
```
Comment (wxiaoguang) [inline tests/integration/repo_fork_test.go]: It can't be a right selector

Comment (lunny) [reply tests/integration/repo_fork_test.go]: image
I found the fork repository item have classes like this, do you have any better idea?

Comment (wxiaoguang) [reply tests/integration/repo_fork_test.go]: `a[href~="/repo-name"]`

Comment (lunny) [reply tests/integration/repo_fork_test.go]: How does that work? We don't know which repositories will be listed.

Comment (wxiaoguang) [reply tests/integration/repo_fork_test.go]: It is in a unit test, why can't we know ......

**Review by wxiaoguang — APPROVED** [review]

**Review by Zettat123 — APPROVED** [review]

**Pull request merged**